### PR TITLE
ML-3377 | Stream Feature View UI Fix

### PR DESCRIPTION
**Pull request by IsraelAgundis**
**Problem**:
Transformations are not being rendered in feast: https://jira.team.affirm.com/browse/ML-3377

**TLDR**:
`StreamFeatureView.from_proto()` has a parameter that skips the deserialization of the `UserDefinedFunctionProto` within a `StreamFeatureViewProto`. This parameter was hard coded to True for stream feature views.

**Explanation**:
UDFs are stored as [dill](https://dill.readthedocs.io/en/latest/)-serialized bytes; dill is designed to load serialized values in the same environment (but not the same machine) they were serialized in. Consequently, loading a UDF from the registry during a new deploy can lead to unexpected behavior. Hence, the current approach of hard coding the parameter in `StreamFeatureView.from_proto()` to True. 

However, as previously stated, the only time that loading UDFs can be problematic is when the environment changes. This would only be a possibility during the execution of `feast apply`, so the hard coded value will be replaced with an additional parameter that determines whether the UDF should be deserialized (because `feast apply` wasn't ran) or shouldn't be deserialized (because `feast apply` was ran).

### CODE REVIEW

File: sdk/python/feast/infra/registry/base_registry.py [hunk 1]
```diff
@@ -248,14 +248,16 @@ def get_stream_feature_view(
...
-        self, project: str, allow_cache: bool = False
+        self, project: str, allow_cache: bool = False, feast_apply_operation: bool = False,
     ) -> List[StreamFeatureView]:
         """
         Retrieve a list of stream feature views from the registry
 
         Args:
             project: Filter stream feature views based on project name
             allow_cache: Whether to allow returning stream feature views from a cached registry
+            feast_apply_operation: Whether a feast apply operation is being executed. Determines whether environment
```
Comment (IsraelAgundis) [inline sdk/python/feast/infra/registry/base_registry.py:259]: Probably can be worded better.

File: sdk/python/feast/infra/registry/base_registry.py [hunk 2]
```diff
@@ -248,14 +248,16 @@ def get_stream_feature_view(
...
+                sensitive commands, such as dill.loads(), are skipped and 'None' is set as their results.
```
Comment (IsraelAgundis) [inline sdk/python/feast/infra/registry/base_registry.py:260]: Probably can be worded better.

File: sdk/python/feast/infra/registry/sql.py [hunk 3]
```diff
@@ -992,7 +992,7 @@ def _list_objects(
                 for row in rows:
                     proto = proto_class.FromString(row[proto_field_name])
                     if python_class == StreamFeatureView:
-                        obj = python_class.from_proto(proto, True if kwargs["skip_udfs"] else False)
+                        obj = python_class.from_proto(proto, kwargs.get("skip_udfs", True))
```
Comment (IsraelAgundis) [inline sdk/python/feast/infra/registry/sql.py:995]: I think the default value should actually be False? If `skip_udfs` isn't passed to this function, then this function was not called from the `feast apply` workflow, hence udfs shouldn't be skipped?

File: sdk/python/feast/infra/registry/base_registry.py [hunk 4]
```diff
@@ -248,14 +248,16 @@ def get_stream_feature_view(
 
     @abstractmethod
     def list_stream_feature_views(
-        self, project: str, allow_cache: bool = False
+        self, project: str, allow_cache: bool = False, feast_apply_operation: bool = False,
```
Comment (Rostifar) [inline sdk/python/feast/infra/registry/base_registry.py:251]: Nice! 

Nit: with respect to the parameter name, we might want to ignore udfs outside of calling feast_apply, so it might help to give it a more general name.

File: sdk/python/feast/infra/registry/sql.py (hunk 3 above)
Comment (Rostifar) [reply sdk/python/feast/infra/registry/sql.py:995]: Right, the default behavior should be to load the udfs
